tests/phpunit/WebTest/Mailing/ValidateBodyMailingComponentTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License along with this program; if not, contact CiviCRM LLC       |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+require_once 'CiviTest/CiviSeleniumTestCase.php';
+/**
+ * Class WebTest_Mailing_ValidateBodyMailingComponentTest 
+ */
+
+class WebTest_Mailing_ValidateBodyMailingComponentTest extends CiviSeleniumTestCase {
+
+  protected function setUp() {
+    parent::setUp();
+  }
+
+  public function testWithoutBodyTextAndHTML() {
+    $this->webtestLogin();
+
+    $this->openCiviPage("admin/component", "action=add&reset=1");
+
+    // fill component name.
+    $componentName = 'ComponentName_' . substr(sha1(rand()), 0, 7);
```

Comment:
Generally, this commit is great in producing a test. :)

Random nitpick... I get antsy whenever there's a truncate on a digest value. In this case (w/`getrandmax()=2^31` and truncated length of 7 hex digits), it turns out to be OK, although it could do better by changing `sha1(...)` to `base_convert()` -- e.g. https://gist.github.com/totten/9beba424caa278d172de4b989c121962